test/test_integration.py

```diff
@@ -0,0 +1,114 @@
+# coding: utf-8
+
+#
+# Copyright 2014-2017 Groupon, Inc.
+# Copyright 2014-2017 The Billing Project, LLC
+#
+# The Billing Project, LLC licenses this file to you under the Apache License, version 2.0
+# (the "License"); you may not use this file except in compliance with the
+# License.  You may obtain a copy of the License at:
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+
+"""
+    Kill Bill
+
+    Kill Bill is an open-source billing and payments platform  # noqa: E501
+
+    OpenAPI spec version: 1.0.0
+    
+    Generated by: https://github.com/swagger-api/swagger-codegen.git
+"""
+
+
+from __future__ import absolute_import
+
+import unittest
+
+import killbill
+from killbill.models.account import Account  # noqa: E501
+from killbill.models.tenant import Tenant  # noqa: E501
+from killbill.models.payment_method import PaymentMethod  # noqa: E501
+from killbill.models.subscription import Subscription  # noqa: E501
+from killbill.models.invoice_dry_run import InvoiceDryRun  # noqa: E501
+from killbill.rest import ApiException
+
+
+
+class TestIntegration(unittest.TestCase):
+    """Account unit test stubs"""
+
+    def setUp(self):
+        pass
+
+    def tearDown(self):
+        pass
+
+    def test_integration(self):
+
+        # Create tenant
+        # TODO WORKING WITH ISSUES*! [ValueError: Invalid value for `api_key`, must not be `None`]
+        tenant = killbill.api.TenantApi()
+        body = Tenant(api_key='bob_5', api_secret='lazar_5')
+        tenant.create_tenant(body, 'test')
+
+        # Get tenant
+        # TODO NOT WORKING! [ApiException: (401) Reason: Unauthorized] or [ValueError: Invalid value for `api_secret`, must not be `None`]
+        body = Tenant(api_key='bob')
+        tenant.get_tenant_by_api_key(body)
+
+        # Upload XML catalog/Fetch XML catalog
+        # TODO NOT WORKING! [ApiException: (404) Reason: Not Found]
+        catalog = killbill.api.CatalogApi()
+        xml_catalog = open("/home/jfrodriguez/Documents/KillBill-Docs/SpyCarBasic.xml", "r+").read()
```

Comment:
Can you check it in, in this directory?

Reply:
Yes it is. If I try `print(xml_catalog)` the console shows:
``` <?xml version="1.0" encoding="UTF-8" standalone="no"?>
<!--
  ~ Copyright 2010-2013 Ning, Inc.
  ~
  ~ Ning licenses this file to you under the Apache License, version 2.0
  ~ (the "License"); you may not use this file except in compliance with the
  ~ License.  You may obtain a copy of the License at:
  ~
  ~    http://www.apache.org/licenses/LICENSE-2.0
  ~
  ~ Unless required by applicable law or agreed to in writing, software
  ~ distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
  ~ WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
  ~ License for the specific language governing permissions and limitations
  ~ under the License.
  -->

<catalog xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:noNamespaceSchemaLocation="CatalogSchema.xsd ">

    <effectiveDate>2013-02-08T00:00:00+00:00</effectiveDate>
    <catalogName>SpyCarBasic</catalogName>
...

Reply:
This test now works, the problem was in the endpoint. `/1.0/kb/catalog/xml` in killbill-0.19.10 the endpoint for that is ` /1.0/kb/catalog`